lnwire/lnwire.go

```diff
@@ -804,7 +804,17 @@ func ReadElement(r io.Reader, element interface{}) error {
 				addrBytesRead += aType.AddrLen()
 
 			default:
-				return &ErrUnknownAddrType{aType}
+				// If we don't understand this address type,
+				// we just ignore it along with the remaining
+				// address bytes.
+				garbage := make([]byte, addrsLen-addrBytesRead)
+				_, err := io.ReadFull(addrBuf, garbage)
+				if err != nil {
+					return err
+				}
+
+				addrBytesRead = addrsLen
```

Comment:
I think we also need to _retain_ those extra bytes, so we still propagate them. Otherwise, with the current logic, the issue CL ran into would still be present as we'd only be storing the address bytes we understand. 

One way to do this would be to create a _new_ address type, which is an `OpaqueAddrs` or w/e, that stores just those raw bytes. This way when we go to write these addrs back on the wire, we write the ones we understand and also the remaining blob. This'll then ensure the addrs propagate as expected.